scripts/run_fcst.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+"""
+The run script for run_fcst.
+"""
+
+import datetime as dt
+import logging
+import os
+import sys
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+
+from uwtools.api.config import get_yaml_config
+from uwtools.api.fs import link as uwlink
+from uwtools.api.fv3 import FV3
+from uwtools.api.logging import use_uwtools_logger
+from uwtools.api.template import render
+from uwtools.api.upp import UPP
+
+
+def _walk_key_path(config, key_path):
+    """
+    Navigate to the sub-config at the end of the path of given keys.
+    """
+    keys = []
+    pathstr = "<unknown>"
+    for key in key_path:
+        keys.append(key)
+        pathstr = " -> ".join(keys)
+        try:
+            subconfig = config[key]
+        except KeyError:
+            logging.error(f"Bad config path: {pathstr}")
+            raise
+        if not isinstance(subconfig, dict):
+            logging.error(f"Value at {pathstr} must be a dictionary")
+            sys.exit(1)
+        config = subconfig
+    return config
+
+
+def link_files(dest_dir, files):
+    """
+    Link a given list of files to the destination directory using the same file names.
+    """
+    for fpath in files:
+        path = Path(fpath)
+        linkname = dest_dir / path.name
+        if linkname.is_symlink():
+            linkname.unlink()
+        logging.info(f"Linking {linkname} -> {path}")
+        linkname.symlink_to(path)
```

Comment:
This might be related to [this](https://github.com/orgs/ufs-community/projects/56/views/10?pane=issue&itemId=76720127&issue=ufs-community%7Cuwtools%7C605), which would require reliable `output()` methods to know what to remove, but this is simpler, as there's only a file or symlink to `rm` -- or, maybe, a `rm -rf` for a directory -- for `fs` targets. Implementing `--reset` for `fs` might be a good start.

On the other hand, I do wonder about mixing behaviors: In some parts of an app's automation, logic (like this) that repeatedly clobbers existing on-disk state; and in other parts (like invocation of drivers) logic that is idempotent. Maybe that's ok, but I wonder if users will understand which part has which behavior.

Reply:
This came in handy during the debugging step when we were trying to get the linking correct. We could add back the idempotence here by just saying if it exists, don't try to re-link. It is external to the rundir though, so this behavior could send users on a wild goose chase through Timbuktu to unlink everything a given SRW script might try to link.

Reply:
> We could add back the idempotence here by just saying if it exists, don't try to re-link.
Would that then be equivalent to using `uw fs link` (CLI or API)? I'm just angling for using `uwtools` functionality where possible, of course. I'll leave it at that -- perfect is not the enemy of good! (And maybe this is even better, I don't know.)